src/fsharp/DotNetFrameworkDependencies.fs

```diff
@@ -25,11 +25,13 @@ module internal FSharp.Compiler.DotNetFrameworkDependencies
             // Use the location of this dll
             location
 
+    let inline ifEmptyUse alternative filename = if String.IsNullOrWhiteSpace filename then alternative else filename
+    
     let getFSharpCoreLibraryName = "FSharp.Core"
     let getFsiLibraryName = "FSharp.Compiler.Interactive.Settings"
     let getDefaultFSharpCoreLocation = Path.Combine(fSharpCompilerLocation, getFSharpCoreLibraryName + ".dll")
     let getDefaultFsiLibraryLocation = Path.Combine(fSharpCompilerLocation, getFsiLibraryName + ".dll")
-    let implementationAssemblyDir = Path.GetDirectoryName(typeof<obj>.Assembly.Location)
+    let implementationAssemblyDir = Path.GetDirectoryName(typeof<obj>.Assembly.Location) |> ifEmptyUse fSharpCompilerLocation
```

Comment:
Could you add additional explanation for that change?

Reply:
in Mono-Wasm `typeof<obj>.Assembly.Location` = `"mscorlib.dll"` with no path and so `Path.GetDirectoryName(typeof<obj>.Assembly.Location)` = `""` which causes an exception later on. In this case I am using `fSharpCompilerLocation` which was resolved in the prior point.

Reply:
@amieres can you put a repro of something that fails on a GitHub repo somewhere.  I certainly want to think about this.

The notion that assemblies are loaded from streams, plays havoc with the a couple of our commonly held invariants.

I think it is a good scenario for us to handle, but I would rather we think about what the solution looks like rather than a few targeted bug fixes.  Although in the end it may just turn out to be soluble that way.

I think probably we might want to look at what fable does too since they may encounter and have solved similar issues.

Reply:
@KevinRansom 
I created a website where different Assemblies for FCS can be tested:
https://amieres.github.io/WASM.html
After the initial load there is an option to select the set of Assemblies to use
with WASM:

- Interpreted FCS 25.0 modified
- Interpreted FCS 33.0
- Interpreted FCS 34.1
- Interpreted FCS 34.1 fmwk

The first one is the only that works it is actually a fork of FCS 25.0 created for this purpose:
https://github.com/fsbolero/FSharp.Compiler.Service

Version 33.0 would work if the 2 issues in this PR were solved (plus a third one, that I don't quite recall right now).

Version 34.0 introduced a new issue by using `AssemblyLoadContext` which is not supported by Mono WASM because according to the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext?view=netstandard-2.1) it is not supposed to be **NetStandard**, but only **NetCore**. OTOH, the package `System.Runtime.Loader` is published in Nuget.org as **.NETStandard 1.5**. I do not know who is right, but the end result is that version 34 wont work with WASM until it goes to NetCore.